app/controllers/githook_controller.rb

```diff
@@ -43,4 +46,9 @@ def authenticate
   rescue ActionController::RedirectBackError
     redirect_to root_path
   end
+
+  def verify_signature(payload_body)
+      signature = 'sha1=' + OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha1'), ENV['SECRET_TOKEN'], payload_body)
+      return halt 500, "Signatures didn't match!" unless Rack::Utils.secure_compare(signature, request.env['HTTP_X_HUB_SIGNATURE'])
```

Comment:
Line is too long. [131/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.